### PR TITLE
feat: Move ai events to spec sidebar

### PR DESCRIPTION
**Pull request by andertun**
The `loading` state is just set to true for now. 

image

Spec button has a loader when it is closed.
image

### CODE REVIEW

File: src/renderer/components/storm/StormBottomPane.tsx [hunk 1]
```diff
@@ -557,7 +557,7 @@ export const StormBottomPane = (props: Props) => {
                     cursor: 'pointer',
                 }}
             >
-                <Menu />
+                <Menu sx={{ '&:hover': { color: 'text.primary', transition: 'color 300ms' } }} />
```
Comment (wejendorp) [inline src/renderer/components/storm/StormBottomPane.tsx:560]: Nice little touch

File: src/renderer/components/storm/context/StormChatContext.tsx [hunk 2]
```diff
@@ -196,21 +192,43 @@ export const StormChatContextProvider = (props: StormChatContextProviderProps) =
...
+        const lastAIMessage = aiConversation.messages.findLast((message) => message.type === 'AI');
+        if (!lastAIMessage) {
+            return [];
+        }
+        return (lastAIMessage as AIMessage).events;
+    }, [aiConversation.messages]);
+
+    // Spec events sidebar
+    const [showSpecs, setShowSpecs] = React.useState(false);
+    const prevSpecEvents = usePrevious(specEvents);
```
Comment (wejendorp) [inline src/renderer/components/storm/context/StormChatContext.tsx:215]: Oh that seems useful, never used that hook before.

File: src/renderer/components/storm/examples/ExamplePrompts.tsx [hunk 3]
```diff
@@ -28,6 +32,24 @@ const examplePrompts: { title: string; prompt: string }[] = [
 export const ExamplePrompts = (props: ExamplePromptsProps) => {
     const { sx, ...boxProps } = props;
 
+    const allPrompts = useMemo<ExamplePrompt[]>(() => {
+        const userPromptsStr = localStorage.getItem('userPrompts');
```
Comment (wejendorp) [inline src/renderer/components/storm/examples/ExamplePrompts.tsx:36]: Pending comment: did we add this to the "admin UI"? aka featureflags box

**Review by wejendorp — APPROVED** [review]
Aside from the filetree copy maybe? lgtm

File: src/renderer/components/storm/examples/ExamplePrompts.tsx (hunk 3 above)
Comment (andertun) [reply src/renderer/components/storm/examples/ExamplePrompts.tsx:36]: No I just added it here for now so we can have custom prompts we like to use to test - will remove it when not needed anymore

File: src/renderer/components/storm/plan-spec/FileTree.tsx [hunk 4]
```diff
@@ -0,0 +1,344 @@
+/**
+ * Copyright 2024 Kapeta Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
```
[the remark this answers is not in the document]
Comment (andertun) [reply src/renderer/components/storm/plan-spec/FileTree.tsx]: No that was a mistake. Removed now f2f06d807961b4e9f245c13ee44054e6297bd5c7

**Pull request merged**